File: DIO.me/Bootcamps & Cursos/04 - Python - Python Backend Developer/02 - Estruturas de Dados/Módulo 02 - Estruturas Condicionais e de Repetição/estruturas_condicionais.py
# - - - - - > Estruturas Condicionais:

#                 As estruturas condicionais são blocos de código que permitem executar diferentes ações 
#            com base em condições específicas. Elas permitem que um programa tome decisões, 
#            avaliando se uma expressão é verdadeira ou falsa e executando diferentes 
#            trechos de código de acordo com o resultado dessa avaliação. 


# - - - - > Exemplos de Estruturas Condicionais Simples:
print("\nExemplos de Estruturas Condicionais Simples:")

idade: int = 20

# - - - > Exemplo de Estrutura Condicional 'if':
print("\nExemplo de Estrutura Condicional 'if':")
if idade >= 18:
    print("Você é maior de idade.")


# - - - > Exemplo de Estrutura Condicional 'if' e 'else':
print("\nExemplo de Estrutura Condicional 'if' e 'else':")
if idade >= 18:
    print("Você é maior de idade.")
else:
    print("Você é menor de idade.")


# - - - > Exemplo de Estrutura Condicional 'if', 'elif' e 'else':
print("\nExemplo de Estrutura Condicional 'if', 'elif' e 'else':")
if idade < 18:
    print("Você é menor de idade.")
elif idade == 18:
    print("Você tem exatamente 18 anos.")
else:
    print("Você é maior de idade.")


# Utilizando operadores lógicos em estruturas condicionais:
print("\nUtilizando operadores lógicos em estruturas condicionais:")
temperatura = 25
if idade >= 18 and temperatura > 20:
    print("Condições ideais para sair.")


# - - - - > Exemplo de Estrutura Condicional Aninhada:
possui_cartao: bool = True
idade: int = 20

print("\nExemplo de Estrutura Condicional Aninhada:")
if idade >= 18:
    print("Você é maior de idade.")

    if possui_cartao:
        print("Você pode fazer compras com desconto.")
    else:
        print("Você pode fazer compras, mas não terá desconto.")
else:
    print("Você é menor de idade.")


# - - - -> Exemplo de Estrutura Condicional Ternária (em uma linha):

resultado = "Aprovado" if idade >= 18 else "Reprovado"

print("\nExemplo de Estrutura Condicional Ternária (em uma linha):")
print(f"Resultado: {resultado}")

File: DIO.me/Bootcamps & Cursos/05 - Java - Desenvolvimento Java/02 - Estruturas de Controle de Fluxo/Módulo 01 - Estruturas Condicionais/estruturas-condicionais/src/edu/dio/me/EstruturasCondicionais.java
package edu.dio.me;

public class EstruturasCondicionais {

    // - - - - - > Estruturas Condicionais:
    //                 Em Java, As estruturas condicionais são usadas para tomar 
    //             decisões baseadas em condições. Elas permitem que diferentes 
    //             blocos de código sejam executados dependendo do resultado de  
    //             uma expressão booleana.

    public static void main(String[] args) {

        // - - - - > Exemplos de Estruturas Condicionais:
        System.out.println("\nExemplos de Estruturas Condicionais:");


        // - - - > Estrutura Condicional Simples:
        System.out.println("\nEstrutura Condicional Simples:");

        int x = 10;

        if (x > 0) {
            System.out.println("'x' é positivo!");
        }


        // - - - > Estrutura Condicional Composta:
        System.out.println("\nEstrutura Condicional Composta:");

        int y = -5;

        if (y > 0) {
            System.out.println("'y' é positivo!");
        } else {
            System.out.println("'y' é negativo ou zero!");
        }


        // - - - > Estrutura Condicional Encadeada:
        System.out.println("\nEstrutura Condicional Encadeada:");

        int z = 0;

        if (z > 0) {
            System.out.println("'z' é positivo!");
        } else if (z < 0) {
            System.out.println("'z' é negativo!");
        } else {
            System.out.println("'z' é zero!");
        }


        // - - - > Estrutura Condicional Ternária:
        System.out.println("\nEstrutura Condicional Ternária:");

        int a = 10;

        String resultado = (a > 0) ? "'a' é positivo!" : (a < 0) ? "'a' é negativo!" : "'a' é zero!";
        System.out.println(resultado);


        // - - - - > Estrutura Condicional Switch Case:
        System.out.println("\nEstrutura Condicional Switch Case:");

        int opcao = 3;
        String tamanho;

        switch (opcao) {
            case 1:
                tamanho = "pequeno";
                break;
            case 2:
                tamanho = "médio";
                break;
            case 3:
                tamanho = "grande";
                break;        
            default:
                tamanho = "padrão";
                break;
        }

        System.out.println(tamanho);
    }
}
